dpnp/backend/extensions/blas/dot.cpp

```diff
@@ -108,24 +107,24 @@ static sycl::event dot_impl(sycl::queue &exec_q,
 }
 
 std::pair<sycl::event, sycl::event> dot(sycl::queue &exec_q,
```

Comment:
`dot`, `dotc` and `dotu` have almost the same code, I would suggest splitting the code to some common function to reduce code duplication, probably something similar to `vm_ext::unary_ufunc` function.

Reply:
Appropriate `_impl` functions are also quite similar.

Reply:
Btw, I don't mind to have that implemented by a separate PR.

Reply:
I will address it in a separate PR.